### PR TITLE
[Feature][ISSUE-2780] support set RestoreMode when starting app

### LINKED ISSUE #2780: [Feature] Support GIC
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

The general log-based incremental checkpoint (GIC) needs the -restoreMode:CLAIM so it can restore from previous savepoint. To support GIC feature, Streampark needs to support -restoreMode  args.

Official example:
bin/flink run -s :savepointPath -restoreMode :mode -n [:runArgs]

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoulii**
## What changes were proposed in this pull request

Issue Number: close #2780

## Verifying this change

- *Manually verified the change by testing locally.*

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala [hunk 1]
```diff
@@ -115,6 +115,7 @@ trait FlinkClientTrait extends Logger {
       flinkConfig.setBoolean(
         SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE,
         submitRequest.allowNonRestoredState)
+      flinkConfig.setString(RestoreMode.RESTORE_MODE, submitRequest.restoreMode.getName);
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala]: Here, we should check the version number of Flink and then decide whether this parameter is needed or not.

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/AppView/StartApplicationModal.vue [hunk 2]
```diff
@@ -89,6 +90,26 @@
...
+        component: 'Select',
+        defaultValue: RestoreModeEnum.NO_CLAIM,
+        componentProps: {
+          options: [
+            { label: 'CLAIM', value: RestoreModeEnum.CLAIM },
+            { label: 'NO_CLAIM', value: RestoreModeEnum.NO_CLAIM },
+            { label: 'LEGACY', value: RestoreModeEnum.LEGACY },
+          ],
+        },
+        afterItem: () =>
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/AppView/StartApplicationModal.vue:109]: Have we checked the version number of Flink here?

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala (hunk 1 above)
Comment (zhoulii) [reply streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala]: done

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/AppView/StartApplicationModal.vue (hunk 2 above)
Comment (zhoulii) [reply streampark-console/streampark-console-webapp/src/views/flink/app/components/AppView/StartApplicationModal.vue:109]: done

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**